src/Symfony/Component/Messenger/EventListener/DispatchSchedulerEventListener.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Psr\Container\ContainerInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageReceivedEvent;
+use Symfony\Component\Scheduler\Event\PostRunEvent;
+use Symfony\Component\Scheduler\Event\PreRunEvent;
+use Symfony\Component\Scheduler\Messenger\ScheduledStamp;
+use Symfony\Component\Scheduler\Messenger\SchedulerTransport;
+
+class DispatchSchedulerEventListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly ContainerInterface $receiverLocator,
+        private readonly ?EventDispatcherInterface $eventDispatcher = null
+    ) {
+    }
+
+    public function onMessageHandled(WorkerMessageHandledEvent $event): void
+    {
+        $envelope = $event->getEnvelope();
+        if (!$scheduledStamp = $envelope->last(ScheduledStamp::class)) {
+            return;
+        }
+
+        $messageContext = $scheduledStamp->messageContext;
+
+        if (!$this->receiverLocator->has($event->getReceiverName())) {
+            return;
+        }
+
+        /** @var SchedulerTransport $schedulerTransport */
+        $schedulerTransport =  $this->receiverLocator->get($event->getReceiverName());
+        $schedule = $schedulerTransport->getMessageGenerator()->schedule();
+
+        $this->eventDispatcher?->dispatch(new PostRunEvent($schedule, $messageContext, $envelope->getMessage()));
```

Comment:
It seems that I would need the same info for pre_run and post_run event. 
I will handle this code duplication.